ShoppingList/ShoppingList/MainWindowViewModel.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Windows.Input;
+
+namespace ShoppingList
+{
+    public class MainWindowViewModel : INotifyPropertyChanged
+    {
+        public event PropertyChangedEventHandler? PropertyChanged;
+
+        private void SetProperty<T>(ref T field, T value, [CallerMemberName]string? propertyName = null)
+        {
+            if (!EqualityComparer<T>.Default.Equals(field, value))
+            {
+                field = value;
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+            }
+        }
+
+        private string _Text;
+        public string Text 
+        { 
+            get => _Text;
+            set => SetProperty(ref _Text, value);
+        }
+
+        private ShoppingItem _SelectedItem;
+        public ShoppingItem SelectedItem
+        {
+            get => _SelectedItem;
+            set => SetProperty(ref _SelectedItem, value);
+        }
+
+        public ICommand AddNewItemCommand { get; }
+
+        public ObservableCollection<ShoppingItem> ShoppingList { get; } = new ObservableCollection<ShoppingItem>();
+
+        public MainWindowViewModel()
+        {
+            AddNewItemCommand = new Command(OnAddNewItem);
+            Text = "";
+            ShoppingList.Add(new ShoppingItem("Bologna"));
+            ShoppingList.Add(new ShoppingItem("Bread"));
+
+            SelectedItem = ShoppingList.First();
+        }
+
+        private void OnAddNewItem()
+        {
+            if(!string.IsNullOrWhiteSpace(Text))
+            {
+                ShoppingList.Add(new ShoppingItem(Text));
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(nameof(ShoppingList)));
+                Text = "";
+            }
```

Comment:
```suggestion
            }
            Text = "";
```
I suggest moving the clearing text option outside of the if statement because if a user puts whitespace into the textbox this will clear that whitespace whenever they try and add a new item (since whitespace isn't allowed it would be best to clear it for the user).